src/xenia/kernel/xam/xam_nui.cc

```diff
@@ -35,23 +57,298 @@ struct X_NUI_DEVICE_STATUS {
+     - 0x0004
+     - 0x0040
+  */
+  uint16_t device_state = cvars::Allow_nui_initialization ? 1 : 0;
+  return device_state >> 1 & 1;
+}
+DECLARE_XAM_EXPORT1(XamNuiIsDeviceReady, kNone, kImplemented);
+
+dword_result_t XamIsNuiAutomationEnabled_entry(unknown_t unk1, unknown_t unk2) {
+  /* Notes:
+     - XamIsNuiAutomationEnabled = XamIsNatalPlaybackEnabled
+     - Always returns X_E_SUCCESS? Maybe check later versions
+     - Recieves param but never interacts with them
+     - No Operations
+  */
+  return X_ERROR_SUCCESS;
+}
+DECLARE_XAM_EXPORT2(XamIsNuiAutomationEnabled, kNone, kStub, kHighFrequency);
+
+dword_result_t XamIsNatalPlaybackEnabled_entry(unknown_t unk1, unknown_t unk2) {
+  /* Notes:
+     - XamIsNuiAutomationEnabled = XamIsNatalPlaybackEnabled
+     - Always returns X_E_SUCCESS? Maybe check later versions
+     - Recieves param but never interacts with them
+     - No Operations
+  */
+  return X_ERROR_SUCCESS;
+}
+DECLARE_XAM_EXPORT2(XamIsNatalPlaybackEnabled, kNone, kStub, kHighFrequency);
+
+dword_result_t XamNuiIsChatMicEnabled_entry() {
+  /* Notes:
+     - calls a second function with a param of uint local_20 [4];
+     - Second function calls ExGetXConfigSetting(7,9,local_30,0x1c,local_40);
+     - Result is sent to *local_20[0] = ^
+     - Once sent back to XamNuiIsChatMicEnabled it looks for byte that
+     correlates to NUI mic setting
+     - return uVar2 = (~(ulonglong)local_20[0] << 0x20) >> 0x23 & 1;
+     - unless the second function returns something -1 or less then
+     XamNuiIsChatMicEnabled 1
+  */
+  return false;
+}
+DECLARE_XAM_EXPORT1(XamNuiIsChatMicEnabled, kNone, kImplemented);
+
+/* HUD Notes:
+   - XamNuiHudGetEngagedTrackingID, XamNuiHudIsEnabled,
+   XamNuiHudSetEngagedTrackingID, XamNuiHudInterpretFrame, and
+   XamNuiHudGetEngagedEnrollmentIndex all utilize the same data address
+   - engaged_tracking_id set second param of XamShowNuiTroubleshooterUI
+*/
+uint32_t nui_unknown_1 = 0;
+uint32_t engaged_tracking_id = 0;
+char nui_unknown_2 = '\0';
+
+dword_result_t XamNuiHudSetEngagedTrackingID_entry(dword_t id) {
+  if ((id & 0xffffffff) == 0) {
+    return static_cast<uint32_t>(id);
+  }
+
+  if (nui_unknown_1 != 0) {
+    engaged_tracking_id = id;
+    return 0;
+  }
+
+  return X_E_FAIL;
+}
+DECLARE_XAM_EXPORT1(XamNuiHudSetEngagedTrackingID, kNone, kImplemented);
+
+qword_result_t XamNuiHudGetEngagedTrackingID_entry() {
+  if (nui_unknown_1 != 0) {
+    return engaged_tracking_id;
+  }
+
+  return 0;
+}
+DECLARE_XAM_EXPORT1(XamNuiHudGetEngagedTrackingID, kNone, kImplemented);
+
+dword_result_t XamNuiHudIsEnabled_entry() {
+  /* Notes:
+     - checks if XamNuiIsDeviceReady false, if nui_unknown_1 exists, and
+     nui_unknown_2 is equal to null terminated string
+     - only returns true if one check fails and allows for other NUI functions
+     to progress
+  */
+  bool result = XamNuiIsDeviceReady_entry();
+  if (nui_unknown_1 != 0 && nui_unknown_2 != '\0' && (result == true)) {
```

Comment:
No need for brackets around ``result == true``. Plus ``&& result`` here is enough